### DISCUSSION TITLE
Highlighting for different checkbox item states

### DISCUSSION THREAD

**Discussion by levouh** [nvim-orgmode/orgmode#126] [Ideas]
Would it be possible to have different highlighting (with the `tree-sitter` branch) for checkbox items that are in different states? It looks like they all resolve to the same node per the parser (see [here](https://github.com/milisims/tree-sitter-org/blob/dda23ecf900261f07843aac7b1f953d827554b56/grammar.js#L448)).

From this then, this would have to be supported by the upstream parser (and resolves to different node "types"/names?) in order for it to work here, right?

**Comment by kristijanhusak** [comment]
Sorry for late response, not following on discussions too much.
This can be easily done via custom highlights (when tree-sitter highlighting is enabled):

```
((checkbox) @check (#eq? @check "\[-\]")) @WarningMsg
((checkbox) @check (#eq? @check "\[ \]")) @ErrorMsg
((checkbox) @check (#eq? @check "\[[xX]\]")) @String
```

We could add support for this, I'm just not sure which hl groups to use by default.

**Reply by levouh** [reply]
Cool, will mess around with this a bit then. They can probably just be linked to the current highlight group (I believe just `Normal`) and users can override them as needed.

@kristijanhusak it looks like these would just go in the `highlights.scm` file then? Messed around with it for all of 2 minutes and didn't have any luck with blindly pasting them in there, but I will mess with it more tomorrow.

**Reply by kristijanhusak** [reply]
@levouh make sure you remove `disable = {'org'}` from your treesitter config, since this relies on TS highlights.
Having this would do the trick:
```
((checkbox) @check (#eq? @check "\[-\]")) @OrgCheckboxHalfChecked
((checkbox) @check (#eq? @check "\[ \]")) @OrgCheckboxUnchecked
((checkbox) @check (#eq? @check "\[[xX]\]")) @OrgCheckboxChecked
```
And then people can easily override it with for example `hi link OrgcheckboxUnchecked ErrorMsg`
